cli/src/program.rs

```diff
@@ -2657,6 +2640,56 @@ fn do_process_program_upgrade(
     Ok(config.output_format.formatted_string(&program_id))
 }
 
+// Attempts to look up the program data account, and adds an extend program data instruction if the
+// program data account is too small.
+fn extend_program_data_if_needed(
+    initial_instructions: &mut Vec<Instruction>,
+    rpc_client: &RpcClient,
+    commitment: CommitmentConfig,
+    fee_payer: Pubkey,
+    program_id: &Pubkey,
+    program_len: usize,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let program_data_address = get_program_data_address(program_id);
+
+    let Some(program_data_account) = rpc_client
+        .get_account_with_commitment(&program_data_address, commitment)?
+        .value
+    else {
+        // Program data has not been allocated yet.
+        return Ok(());
+    };
+
+    let required_len = UpgradeableLoaderState::size_of_programdata(program_len);
+    if required_len > usize::try_from(MAX_PERMITTED_DATA_LENGTH).unwrap() {
+        return Err(format!(
+            "New program ({program_id}) data account is too big: {required_len}.\n\
+             Maximum program size: {MAX_PERMITTED_DATA_LENGTH}.",
+        )
+        .into());
+    }
+
+    let current_len = program_data_account.data.len();
+    let Some(additional_bytes) = required_len.checked_sub(current_len) else {
+        // Current allocation is sufficient.
+        return Ok(());
+    };
+
+    if additional_bytes == 0 {
+        return Ok(());
+    }
```

Comment:
Nit: I think `saturating_sub` is actually more straightforward here, since the current allocation is sufficient in both cases.

Reply:
Good point.  Done.